sdk/android/src/java/org/webrtc/Camera2Session.java

```diff
@@ -404,12 +406,22 @@ private void reportError(String error) {
     }
   }
 
+  @Override
+  public void setOrientation(int orientation) {
+    checkIsOnCameraThread();
+    Logging.d(TAG, "Set Orientation: " + orientation);
+
+    // The device orientation is locked to portrait in Signal clients.
+    // Instead of using CameraSession.getDeviceOrientation(), rely
+    // on the client to indicate the current rotated orientation.
+    deviceOrientation = orientation;
+  }
+
   private int getFrameOrientation() {
-    int rotation = CameraSession.getDeviceOrientation(applicationContext);
```

Comment:
Would it make sense to have the this.deviceOrientation be an Int = null when is setOrientation is not called?  Then if it's not called, we'll still have the old behavior of getting it from the current state of the app.  As you have it now, it think it will get "stuck" on the initial orientation if setOrientation isn't called.  

I'm thinking of a future when clients want to not be portrait locked and not call setOrientation.

Reply:
Okay, I updated to pass in @Nullable Integer, so setOrientation() can turn default behavior on if null is passed.